File: easy-lowcode-frontend/src/api/apiManagement.ts
import request from '@/utils/request'

export interface ApiManagement {
  id?: string | number
  apiName: string
  apiPath: string
  apiMethod: string
  apiType: string
  sourceId?: number
  description?: string
  requestConfig?: string
  responseConfig?: string
  status?: number
  version?: string
  authRequired?: number
  rateLimit?: number
  sortOrder?: number
  createTime?: string
  updateTime?: string
}

/**
 * 分页查询API列表
 */
export function getApiPage(params: {
  current: number
  size: number
  apiName?: string
  apiType?: string
  status?: number
}): Promise<any> {
  return request({
    url: '/collector/api-management/page',
    method: 'get',
    params,
  })
}

/**
 * 获取API详情
 */
export function getApiById(id: number | string): Promise<ApiManagement> {
  return request({
    url: `/collector/api-management/${id}`,
    method: 'get',
  })
}

/**
 * 注册外部接口API
 */
export function registerExternalApi(data: ApiManagement): Promise<void> {
  return request({
    url: '/collector/api-management/register-external',
    method: 'post',
    data,
  })
}

/**
 * 更新API信息
 */
export function updateApi(id: number | string, data: ApiManagement): Promise<void> {
  return request({
    url: `/collector/api-management/${id}`,
    method: 'put',
    data,
  })
}

/**
 * 删除API
 */
export function deleteApi(id: number | string): Promise<void> {
  return request({
    url: `/collector/api-management/${id}`,
    method: 'delete',
  })
}

/**
 * 批量删除API
 */
export function batchDeleteApi(ids: (number | string)[]): Promise<void> {
  return request({
    url: '/collector/api-management/batch',
    method: 'delete',
    data: ids,
  })
}

/**
 * 启用/禁用API
 */
export function updateApiStatus(id: number | string, status: number): Promise<void> {
  return request({
    url: `/collector/api-management/${id}/status`,
    method: 'put',
    params: { status },
  })
}

File: easy-lowcode-collector/src/main/java/com/dabai/easy_lowcode/collector/entity/ApiManagement.java
package com.dabai.easy_lowcode.collector.entity;

import com.baomidou.mybatisplus.annotation.TableField;
import com.baomidou.mybatisplus.annotation.TableName;
import com.dabai.easy_lowcode.database.entity.BaseEntity;
import lombok.Data;
import lombok.EqualsAndHashCode;

/**
 * API管理实体
 */
@Data
@EqualsAndHashCode(callSuper = true)
@TableName("collector_api_management")
public class ApiManagement extends BaseEntity {
    
    /**
     * API名称
     */
    private String apiName;
    
    /**
     * API路径
     */
    private String apiPath;
    
    /**
     * HTTP方法 (GET/POST/PUT/DELETE/PATCH)
     */
    private String apiMethod;
    
    /**
     * API类型 (TABLE_RESOURCE-表资源, EXTERNAL-外部接口)
     */
    private String apiType;
    
    /**
     * 来源ID (表资源ID或外部接口配置ID)
     */
    private Long sourceId;
    
    /**
     * API描述
     */
    private String description;
    
    /**
     * 请求配置JSON (参数、headers等)
     */
    private String requestConfig;
    
    /**
     * 响应配置JSON (字段映射等)
     */
    private String responseConfig;
    
    /**
     * 状态 (0-禁用 1-启用)
     */
    private Integer status = 1;
    
    /**
     * API版本
     */
    private String version = "v1";
    
    /**
     * 是否需要认证 (0-否 1-是)
     */
    private Boolean authRequired = false;
    
    /**
     * 限流次数/分钟
     */
    private Integer rateLimit;
    
    /**
     * 排序号
     */
    private Integer sortOrder = 0;
    
    /**
     * 数据源名称（非数据库字段，用于展示）
     */
    @TableField(exist = false)
    private String datasourceName;
    
    /**
     * 表名（非数据库字段，用于展示）
     */
    @TableField(exist = false)
    private String tableName;
}
